RELEASE.md

```diff
@@ -27,27 +27,47 @@ work. Here's one example workflow (assuming of course you have rights to
 directly push to the upstream repo).
 
 Create a temporary clone of the repo, e.g. to `/tmp`. Then create the `tag` and
-push to origin. For example, to create a release called `v0.24.0`, you would run
-the following commands:
+push to origin. Despite Knative 1.0, the release tags in this repo are still versioned in the `v0.x.y` format and correspond to the Knative tags `knative-v1.x.y`.
+This starts from `v0.27.0` == `knative-v1.0.0` and each MINOR version addition in the `knative-v1.x.y` tag corresponds to the same addition on the `0.x.z` tag.
+
+PATCH versions are ignored in the branch name. In case a release version MUST be patched, just redo the release in the same release branch pointing to the correct commit and with the correct tag, i.e:
+
+| Branch      | Normal Tag  | Knative Tag      |
+|-------------|-------------|------------------|
+| release-1.1 | `v0.27.0`   | `knative-v1.0.x` |
+| release-1.1 | `v0.28.0`   | `knative-v1.1.x` |
+| release-1.1 | `v0.28.1`   | `knative-v1.1.x` |
+| release-1.2 | `v0.29.0`   | `knative-v1.2.x` |
+| release-1.3 | `v0.30.0`   | `knative-v1.3.x` |
+and so on...
+
+As a practical example, to create a release called `v0.27.0` you would run the following commands:
 
 ```shell
 cd /tmp
 git clone git@github.com:vmware-tanzu/sources-for-knative.git
 cd sources-for-knative
 
 # checkout the branch created above
-git checkout release-0.24
+git checkout release-1.0
 
 # associate the tag with the branch
-git tag -a v0.24.0 -m "Release v0.24.0"
-git push origin v0.24.0
+git tag -a v0.27.0 -m "Release v0.27.0"
+
+# since knative version 1.0 we need to create an aditional tag that matches the
+# knative-v1.x.y tag format
+git tag -a knative-v1.0.0 -m "Release knative-v1.0.0"
+
+git push origin v0.27.0
```

Comment:
nit: I would make this release and the knative tag env variables since it's easy to mess this up when copying/pasting doing manual.

We usually follow this pattern (in other non-kn repos) to ease releasing for the operator, e.g. in [`govmomi`](https://github.com/vmware/govmomi/blob/master/RELEASE.md): 

```console
RELEASE=v0.27.0
KN_RELEASE=knative-v1.0.0
```

and then reuse these across the `tag` and `push` commands, e.g.

```console
git tag -a ${RELEASE} -m "Release ${RELEASE}"
git tag -a ${KN_RELEASE} -m "Release ${KN_RELEASE}"

git push origin ${RELEASE}
git push origin ${KN_RELEASE}
```